src/storage/mod.rs

```diff
@@ -903,18 +959,29 @@ impl<E: Engine, L: LockManager> Storage<E, L> {
         let res = self.read_pool.spawn_handle(
             async move {
                 metrics::tls_collect_command_count(CMD, priority_tag);
+                {
+                    let mut key = vec![];
+                    if let Some(end_key) = &end_key {
+                        key = end_key.to_vec();
+                    }
+                    if reverse_scan {
```

Comment:
There's much code judging `reverse_scan`, maybe we can ignore it here and judge it when splitting.